### ISSUE TITLE
Support fat-jar deployments

### ISSUE THREAD

**Issue by adamw** [heroku/sbt-heroku#21]
Right now it's only possible to deploy applications build using `sbt-native-packager`. Another popular plugin for building release artifacts is `sbt-assembly`, which creates a single, executable fat jar.

With a simple modification the existing plugin may support such scenarios as well. Would you be interested in such a feature? I can try a PR, but would be good to know if it has any chances of being accepted :)

**Comment by jkutner** [comment]
Yea, that's definitely something I'd like to support.

**Comment by jkutner** [comment]
I'm still not sure of the best way to support sbt-assembly natively. But you can get it working with a "stage" task by adding this to your `build.sbt` (note that this *requires* version 0.4.3.1 of sbt-heroku or higher).

```scala
val stage = taskKey[Unit]("Stage task")

val Stage = config("stage")

stage := {
  (assembly in Compile).value
  val stageDir = baseDirectory.value / "target" / "universal" / "stage"
  IO.createDirectory(stageDir)
  IO.copyFile((assemblyOutputPath in assembly).value, stageDir / s"${name.value}.jar")
}

herokuProcessTypes in Compile := Map(
  "web" -> "java -Dhttp.port=$$PORT -jar target/universal/stage/*.jar"
)
```

And of course this in `project/plugins.sbt`

```scala
addSbtPlugin("com.heroku" % "sbt-heroku" % "0.4.3.1")
```

**Comment by adamw** [comment]
Yeah but that's "hacking" the plugin a bit; there's a hard requirement now for a file called target/universal/stage. In fact currently I just copied the plugin's source and removed that requirement, leaving most of other things intact.

Btw., the root-project-only detection is also broken (at least for my project) as I have a `project` subdirectory in each module.

I wanted to spend some time on this but got some more work unfortunately, I'll try to get back soon :)

**Comment by jkutner** [comment]
Yea, my example is only intended as a workaround. I'm trying to find a way to auto-detect the use of the sbt-assembly plugin, but I don't want to avoid the dir check` when the sbt-native-packager` plugin is in use. I need to make sure it fails nicely if those folks forget to run `stage`.

I'll probably add a way to explicitly tell the plugin that sbt-assembly is being used.

**Comment by jkutner** [comment]
Fixed in fbdff49b9834d50cebcb45c25782c9a07f1342d9. should be available in 0.5.2
